jsonpickle/ext/pandas.py

```diff
@@ -161,6 +161,11 @@ def restore(self, data):
         buf, meta = self.pp.restore_pandas(data)
         dtype = meta.get('dtype', None)
         name_bundle = {k: v for k, v in meta.items() if k in {'name', 'names'}}
+        if 'names' in name_bundle:
+            # pandas won't accept the names arg in a future version
+            names = name_bundle.pop('names')
+            # name arg must be hashable so cast to tuple first
+            name_bundle['name'] = tuple(names)
```

Comment:
Thanks for these updates, this is really good stuff.

I'm not sure if this would help at all, but this block of code made me wonder, "can we maybe save one dict lookup by trying to `pop()` unconditionally and then catching the KeyError there?

eg...

```
try:
    names = name_bundle.pop('names')
    ...
except KeyError:
    pass  # 'names' is not present in name_bundle
```

Pretty trivial and this is certainly fine as-is, but sometimes every little dict lookup counts.

Reply:
Oh whoops, this wasn't supposed to be in this commit, I guess I forgot to clean my work tree! Anyway, sure, I can benchmark that change. `try/except` blocks aren't "zero-cost" until CPython 3.11 though, so if it has to catch an error in a lower version that could slow it down a bit.